pkg/agent/controller/networkpolicy/reject.go

```diff
@@ -299,29 +306,94 @@ func getRejectOFPorts(rejectType RejectType, sIface, dIface *interfacestore.Inte
+					AddLoadRegMark(openflow.AntreaFlexibleIPAMRegMark).AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
+		} else {
+			mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
+				return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
+					AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
 		}
 	case RejectLocalToRemote:
 		tableID := openflow.L3ForwardingTable.GetID()
 		// L3ForwardingTable is not initialized for ExternalNode case since layer 3 is not needed.
 		if nodeType == config.ExternalNode {
 			tableID = openflow.L2ForwardingCalcTable.GetID()
 		}
-		mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
-			return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
-				AddResubmitAction(nil, &tableID)
+		if isAntreaFlexibleIPAMSrc {
+			mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
+				return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
+					AddLoadRegMark(openflow.AntreaFlexibleIPAMRegMark).AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
+		} else {
+			mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
+				return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
+					AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
+		}
+	case RejectServiceRemoteToLocal:
+		if isAntreaFlexibleIPAMDst {
+			tableID := openflow.ConntrackTable.GetID()
+			mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
+				return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
+					AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
 		}
 	}
 	return mutatePacketOut
 }
+
+func parseFlexibleIPAMStatus(pktIn *ofctrl.PacketIn, nodeConfig *config.NodeConfig, srcIP string, srcFound bool, dstIP string, dstFound bool) (bool, bool, uint32, bool, error) {
+	// isAntreaFlexibleIPAMSrc is true if srcIP is local FlexibleIPAM Pod.
+	isAntreaFlexibleIPAMSrc := false
+	// isAntreaFlexibleIPAMDst is true if dstIP is local FlexibleIPAM Pod.
+	isAntreaFlexibleIPAMDst := false
+	// ctZone is not zero if FlexibleIPAM is enabled.
+	var ctZone uint32
+	if srcFound && nodeConfig.PodIPv4CIDR != nil && !nodeConfig.PodIPv4CIDR.Contains(net.ParseIP(srcIP)) {
+		isAntreaFlexibleIPAMSrc = true
+	}
+	if dstFound && nodeConfig.PodIPv4CIDR != nil && !nodeConfig.PodIPv4CIDR.Contains(net.ParseIP(dstIP)) {
+		isAntreaFlexibleIPAMDst = true
+	}
+	matches := pktIn.GetMatches()
+	if match := getMatchRegField(matches, openflow.CtZoneField); match != nil {
```

Comment:
Could you add some comments to explain this block? I do not understand from the code.

Reply:
Added comment here.
This is to get the ctZone from incoming packet, and ctZone will be set to reject packet, otherwise conntrack cannot work properly, because we use one ctZone per VLAN.